src/tests/ftest/scrubber/aggregation.py

```diff
@@ -0,0 +1,70 @@
+"""
+  (C) Copyright 2018-2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+
+from scrubber_test_base import TestWithScrubber
+from telemetry_utils import TelemetryUtils
+
+
+class TestScrubberEvictWithAggregation(TelemetryUtils, TestWithScrubber):
+    # pylint: disable=too-many-ancestors
+    """Inject Checksum Fault with scrubber enabled
+    and scrubber threshold set to a certain value.
+    Aggregation is run on the background.
+
+    :avocado: recursive
+    """
+    def test_target_eviction_during_aggregation(self):
+        """JIRA ID: DAOS-7333
+
+        1. Start the Aggregation task.
+        2. Create checksum faults above scrubber threshold
+        and see whether SSD auto eviction works as expected.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=scrubber,faults
+        :avocado: tags=TestScrubberEvictWithAggregation,test_target_eviction_during_aggregation
+        """
+        initial_metrics = {}
+        final_metrics = {}
+        self.add_pool()
+        # Disable the aggregation on the pool.
+        self.pool.set_property("reclaim", "disabled")
+        self.add_container(self.pool)
+        # Pool and Containers are already created. Just run the IOR.
+        self.run_ior_with_pool(create_cont=False)
+        initial_aggregation_metrics = self.get_metrics("engine_pool_vos_aggregation_obj_scanned")
+        # Enable the aggregation on the pool.
+        self.pool.set_property("reclaim", "time")
+        # Now enable the scrubber on the pool.
+        self.pool.set_prop(properties="scrub:timed,scrub-freq:1,scrub-thresh:3")
+        initial_metrics = self.scrubber.get_scrub_corrupt_metrics()
+        self.run_ior_and_check_scruber_status(pool=self.pool, cont=self.container)
+        # We want both aggregation and scrubber tasks
+        # to run in parallel during this time.
+        start_time = 0
+        finish_time = 0
+        start_time = time.time()
+        while int(finish_time - start_time) < 120:
```

Comment:
You may want to look at https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/general_utils.py#L1459 which could be used here like so:
```
        status = wait_for_result(
            self.log, self.verify_scrubber_metrics_value, 120, delay=10,
            initial_metrics=initial_aggregation_metrics, final_metrics=final_aggregation_metrics)
```

Reply:
@phender : I don't think wait_for_result can be used here. We are using two methods inside the loop. One for getting the metrics for aggregation and another for verifying the value with the initial metrics... Does wait for result support running two methods.

Reply:
To use `wait_for_result`, you would have to define a helper method that calls both `get_metrics` and `verify_scrubber_metrics_value`. Personally, I think that makes the code harder to follow